runtime/gc_base/modronapi.cpp

```diff
@@ -878,10 +879,17 @@ j9gc_set_allocation_sampling_interval(J9VMThread *vmThread, UDATA samplingInterv
 {
 	MM_GCExtensions *extensions = MM_GCExtensions::getExtensions(vmThread->javaVM);
 	if (0 == samplingInterval) {
-		/* avoid (env->_oolTraceAllocationBytes) % 0 which could be undefined. */
+		/* avoid (env->_traceAllocationBytes) % 0 which could be undefined. */
 		samplingInterval = 1;
 	}
 	extensions->oolObjectSamplingBytesGranularity = samplingInterval;
+	if ((UDATA_MAX != samplingInterval) && !extensions->disableInlineAllocationForSamplingBytesGranularity) {
+		extensions->disableInlineAllocationForSamplingBytesGranularity = true;
+		MM_EnvironmentBase *env = MM_EnvironmentBase::getEnvironment(vmThread->omrVMThread);
+		env->_objectAllocationInterface->disableCachedAllocations(env);
```

Comment:
does this mean only current thread (`MM_EnvironmentBase`) `disableCachedAllocations`? how about other threads?
`j9gc_set_allocation_sampling_interval` won't be called for each thread.

Reply:
the _objectAllocationInterface is global, if we call disableCachedAllocations(), it is for all of threads.

Reply:
> the _objectAllocationInterface is global, if we call disableCachedAllocations(), it is for all of threads.

Is it correct? I believe we create OAI per each thread: https://github.com/eclipse/omr/blob/eb71465140b3b7fd74f8ad12a65da4d384aa945e/gc/base/Configuration.cpp#L190

Reply:
yes, you are right, so should we keep different sampling interval for each thread?

Reply:
I don't know this code good enough for now. For first look it should be applicable to all threads

Reply:
> so should we keep different sampling interval for each thread?

`j9gc_set_allocation_sampling_interval` is not called for each thread.
Though this is GC internal, the application doesn't expect different intervals for different threads, the sampling events might not be fired strictly accordingly to the intervals, in the long run, statistically the number of sampling event is expected to match the allocation/intervals specified.

Reply:
> I don't know this code good enough for now. For first look it should be applicable to all threads

we need to handle disableCachedAllocations() for whole threads and any new thread need to monitor extensions->disableInlineAllocationForSamplingBytesGranularity for initializing _cachedAllocationsEnabled

Reply:
So looks like we need global flag (for example in GC Extensions) to check it every time when we executing allocation path. Is it correct?

Reply:
we have disableInlineAllocationForSamplingBytesGranularity and oolObjectSamplingBytesGranularity in GC Extensions already, they have been checked in allocation path.